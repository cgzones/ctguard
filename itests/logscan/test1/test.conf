logscan {
  logfile "test.log" {
  }

  logfile "test_timeout.log" {
    timeout_warning = 2
  }

  state_file = "test.state"
  output_kind = file
  output_path = "test.output"

  systemd_input = false
}
