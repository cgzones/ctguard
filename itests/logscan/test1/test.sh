#!/bin/sh

set -eu

cleanup () {
  rm -f test.log test_timeout.log test.output test.pid test.state
}

cleanup

touch test.log test_timeout.log test.state

../../../ctguard-logscan --cfg-file test.conf --unittest -f &
pid=$!
echo "daemon running with pid ${pid}\n"

sleep 1

#test normale logs
echo "test" >> test.log
echo "test2" >> test.log
echo "test" >> test_timeout.log

sleep 2

#test truncation1
echo "" > test.log
echo "test3" >> test.log

sleep 2

#test truncation2
echo "" > test.log
sleep 2
echo "test4" >> test.log

sleep 2

#test remove
rm test.log
sleep 2
echo "test5" >> test.log


sleep 2

pid2=`pidof ctguard-logscan`
if [ "X${pid}" != "X${pid2}" ]; then
  echo "daemon not running anymore!!\n${pid} vs ${pid2}\nFAILURE!!\n\n";
  exit 1
fi

kill -INT ${pid}

sleep 2

echo "Comapring expected vs actual output:"
diff -u test.output test.output.expected

cleanup

echo "\n\n\nsuccess!!!\n\n"
