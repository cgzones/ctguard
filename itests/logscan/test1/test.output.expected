 [daemon] : ctguard-logscan started
 [test.log] : test
 [test.log] : test2
 [test_timeout.log] : test
 [test.log] : !File truncated
 [test.log] : test3
 [test_timeout.log] : !Timeout alert
 [test.log] : !File truncated
 [test.log] : test4
 [test.log] : !File went down
 [test.log] : !File coming alive
 [test.log] : !File got replaced
 [test.log] : !File truncated
 [test.log] : test5
 [daemon] : ctguard-logscan shutting down...
